Test DB migrations with Percona XtraDB MySQL server 5.7.25

lucasmrod opened the pull request:
#15881

This PR adds a script to test DB migrations with Percona XtraDB 5.7.25.

PS: To run this test before we merge this PR to `main` you will need to change step 2 (`Make sure to be on latest main`), instead of `main` use this branch `15881-test-migrations-with-percona`.

sabrinabuckets commented:
Hey @lucasmrod can you clarify for me the expected behavior on this one? I pulled the branch & ran the script, but it got stuck on a loop of 
```
mysql --host 127.0.0.1 --port 3306 -uroot -e 'SELECT 1=1;'
./tools/percona/test/upgrade.sh: line 23: mysql: command not found
+ echo 'Waiting for Percona XtraDB MySQL Server...'
Waiting for Percona XtraDB MySQL Server...
+ sleep 10
```

And it reset my DB in the process.

lucasmrod commented:
> Hey @lucasmrod can you clarify for me the expected behavior on this one? I pulled the branch & ran the script, but it got stuck on a loop of
> 
> ```
> mysql --host 127.0.0.1 --port 3306 -uroot -e 'SELECT 1=1;'
> ./tools/percona/test/upgrade.sh: line 23: mysql: command not found
> + echo 'Waiting for Percona XtraDB MySQL Server...'
> Waiting for Percona XtraDB MySQL Server...
> + sleep 10
> ```
> 
> And it reset my DB in the process.

Hi @sabrinabuckets!

Really sorry about this (resetting the DB in the process). I've added more notes to the test to clarify the test is destructive and added steps to backup and restore.

PS: To run this test before we merge this PR to `main` you will need to change step 2 (`Make sure to be on latest main`), instead of `main` use this branch `15881-test-migrations-with-percona`.

lucasmrod commented:
> When I went to switch back to main I got this:
bri@Bris-MBP-2 fleet % git switch main
Already on 'main'

This is expected because the test will checkout to latest `main` to confirm if all DB migrations are compatible with Percona Server.

Sampfluger88 commented:
@lucasmrod multiple PRs changing the smoke-test.md small conflict here.

Sampfluger88 reviewed (approved):
Issue template 👍

xpkoala commented:
@lucasmrod Completed an upgrade using the scripts provided in this branch and did some light smoke testing. This can be merged to `main`.

xpkoala reviewed (approved):
I was able to successfully use the update scripts and smoke test Fleet after running the migration tests.

ksatter commented:
@xpkoala Can you make certain that this covers an upgrade from v4.42.0 to v4.44.0?

xpkoala commented:
@ksatter I'll do that now, I only attempted 4.43.0 >> 4.44.0. I'll update in just a moment.

lucasmrod commented:
The added script automatically tests migrating from v4.42.0 -> latest.

xpkoala commented:
@ksatter I was under a false assumption on how the test scripts were working and assumed it started the upgrade process from my locally checked out branch. I got confirmation from Lucas that the script actually starts at 4.42.0, so this case has been run.

ksatter commented:
Thanks for checking @xpkoala!
